libsupport/include/katana/ArrowVisitor.h

```diff
@@ -73,6 +73,12 @@ VisitArrowCast(const arrow::Array& array) {
 
 inline arrow::Type::type
 GetArrowTypeID(const arrow::ArrayBuilder* builder) {
+  if (builder->type()->id() == arrow::Type::EXTENSION) {
+    auto et = std::dynamic_pointer_cast<arrow::ExtensionType>(builder->type());
```

Comment:
This dynamic cast should always be correct, I wrote it as a static_cast in my version :) #909

Reply:
Perfect, Ill close this